src/Components/CounterButtons.js

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+function CounterButtons({counter, setCounter, fruitName}) {
+
+    return (
+        <article className="fruit-buttons">
+            <h2>{fruitName}</h2>
+            <div>
+                <button
+                    type="button"
+                    onClick={() => setCounter(counter > 0 ? counter - 1 : counter)}
+                >
+                    -
+                </button>
+                <p>{counter}</p>
+                <button
+                    type="button"
+                    onClick={() => setCounter(counter + 1)}
```

Comment:
Wellicht gaat dit nu iets te ver, maar doordat je `counter` gebruikt in deze functie zorgt dit bij een verandering van de waarde van `counter` dat deze functie telkens voor een onnodige rerender zorgt. Je kan dit tegen gaan door het volgende te doen:
```suggestion
                    onClick={() => setCounter(curr + 1)}
```

`curr` is in dit voorbeeld de huidige waarde van `current`, dezelfde waarde dus. Het werkt hetzelfde, maar zorgt niet voor een onnodige rerender.

Het is niet heel erg als je dat rerender-gebeuren nog niet echt doorhebt of kan onthouden, maar ik wil er je wel op wijzen 😉 . In essentie gaat het er dus om dat wanneer je iets met die set-functie wil doen van de useState, dan kun je de huidige waarde op deze manier gebruiken om een nieuwe waarde te setten.